### ISSUE TITLE
Install via pip causes all version of semgrep to be downloaded

### ISSUE THREAD

**Issue by anotherbridge** [semgrep/semgrep#7032]
**Describe the bug**
When installing semgrep in a freshly created virtual environment via pip all the package versions until 0.82.0 will be downloaded. This causes semgrep to use unnecessary space and also prolongs builds for a significant amount of time. Moreover, it will install version 0.82.0 instead of 1.8.0

Yet, when a previous version of semgrep is installed and an update is performed no issues occur. 

**To Reproduce**
```
python3 -m venv .venv
source .venv/bin/activate
pip install semgrep 
```

**Expected behavior**
Only the latest version, i.e. 1.8.0 should be downloaded and installed. 

**What is the priority of the bug to you?**

- [x] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Environment**
This occured on a Debian machine as well as on a Mac having an ARM processor.

**Comment by samueljsb** [comment]
I can't reproduce this myself:

```console
$ python --version --version
Python 3.8.15 (default, Oct 11 2022, 21:52:37)
[Clang 14.0.0 (clang-1400.0.29.102)]
$ python -m venv venv
$ venv/bin/pip install semgrep --no-cache
Collecting semgrep
  Downloading semgrep-1.8.0-cp37.cp38.cp39.py37.py38.py39-none-macosx_10_14_x86_64.whl (23.5 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 23.5/23.5 MB 9.3 MB/s eta 0:00:00
Collecting glom~=22.1
  Downloading glom-22.1.0-py2.py3-none-any.whl (100 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100.7/100.7 KB 11.2 MB/s eta 0:00:00
Collecting jsonschema~=4.6
  Downloading jsonschema-4.17.3-py3-none-any.whl (90 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.4/90.4 KB 12.8 MB/s eta 0:00:00
Collecting ruamel.yaml<0.18,>=0.16.0
  Downloading ruamel.yaml-0.17.21-py3-none-any.whl (109 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 109.5/109.5 KB 12.1 MB/s eta 0:00:00
Collecting colorama~=0.4.0
  Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
Collecting requests~=2.22
  Downloading requests-2.28.2-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.8/62.8 KB 51.3 MB/s eta 0:00:00
Collecting tomli~=2.0.1
  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting packaging~=21.0
  Downloading packaging-21.3-py3-none-any.whl (40 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.8/40.8 KB 116.1 MB/s eta 0:00:00
Collecting click~=8.1
  Downloading click-8.1.3-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.6/96.6 KB 11.9 MB/s eta 0:00:00
Collecting click-option-group~=0.5
  Downloading click_option_group-0.5.5-py3-none-any.whl (12 kB)
Collecting wcmatch~=8.3
  Downloading wcmatch-8.4.1-py3-none-any.whl (39 kB)
Collecting python-lsp-jsonrpc~=1.0.0
  Downloading python_lsp_jsonrpc-1.0.0-py3-none-any.whl (8.5 kB)
Collecting peewee~=3.14
  Downloading peewee-3.15.4.tar.gz (860 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 860.9/860.9 KB 9.7 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting typing-extensions~=4.2
  Downloading typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting tqdm~=4.46
  Downloading tqdm-4.64.1-py2.py3-none-any.whl (78 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.5/78.5 KB 18.0 MB/s eta 0:00:00
Collecting boltons~=21.0
  Downloading boltons-21.0.0-py2.py3-none-any.whl (193 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 193.7/193.7 KB 11.9 MB/s eta 0:00:00
Collecting urllib3~=1.26
  Downloading urllib3-1.26.14-py2.py3-none-any.whl (140 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 140.6/140.6 KB 10.8 MB/s eta 0:00:00
Collecting defusedxml~=0.7.1
  Downloading defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
Collecting parsy~=2.0
  Downloading parsy-2.0-py2.py3-none-any.whl (7.2 kB)
Collecting attrs>=21.3
  Downloading attrs-22.2.0-py3-none-any.whl (60 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.0/60.0 KB 26.5 MB/s eta 0:00:00
Collecting face>=20.1.0
  Downloading face-22.0.0-py3-none-any.whl (54 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.5/54.5 KB 97.1 MB/s eta 0:00:00
Collecting pkgutil-resolve-name>=1.3.10
  Downloading pkgutil_resolve_name-1.3.10-py3-none-any.whl (4.7 kB)
Collecting importlib-resources>=1.4.0
  Downloading importlib_resources-5.10.2-py3-none-any.whl (34 kB)
Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0
  Downloading pyrsistent-0.19.3-cp38-cp38-macosx_10_9_universal2.whl (82 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 82.5/82.5 KB 11.8 MB/s eta 0:00:00
Collecting pyparsing!=3.0.5,>=2.0.2
  Downloading pyparsing-3.0.9-py3-none-any.whl (98 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.3/98.3 KB 10.8 MB/s eta 0:00:00
Collecting ujson>=3.0.0
  Downloading ujson-5.7.0-cp38-cp38-macosx_10_9_x86_64.whl (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.4/57.4 KB 22.3 MB/s eta 0:00:00
Collecting idna<4,>=2.5
  Downloading idna-3.4-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.5/61.5 KB 11.2 MB/s eta 0:00:00
Collecting certifi>=2017.4.17
  Downloading certifi-2022.12.7-py3-none-any.whl (155 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 155.3/155.3 KB 12.4 MB/s eta 0:00:00
Collecting charset-normalizer<4,>=2
  Downloading charset_normalizer-3.0.1-cp38-cp38-macosx_10_9_x86_64.whl (122 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 122.9/122.9 KB 17.4 MB/s eta 0:00:00
Collecting ruamel.yaml.clib>=0.2.6
  Downloading ruamel.yaml.clib-0.2.7-cp38-cp38-macosx_10_9_x86_64.whl (143 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 143.4/143.4 KB 12.2 MB/s eta 0:00:00
Collecting bracex>=2.1.1
  Downloading bracex-2.3.post1-py3-none-any.whl (12 kB)
Collecting zipp>=3.1.0
  Downloading zipp-3.12.0-py3-none-any.whl (6.6 kB)
Using legacy 'setup.py install' for peewee, since package 'wheel' is not installed.
Installing collected packages: peewee, charset-normalizer, boltons, zipp, urllib3, ujson, typing-extensions, tqdm, tomli, ruamel.yaml.clib, pyrsistent, pyparsing, pkgutil-resolve-name, parsy, idna, face, defusedxml, colorama, click, certifi, bracex, attrs, wcmatch, ruamel.yaml, requests, python-lsp-jsonrpc, packaging, importlib-resources, glom, click-option-group, jsonschema, semgrep
  Running setup.py install for peewee ... done
Successfully installed attrs-22.2.0 boltons-21.0.0 bracex-2.3.post1 certifi-2022.12.7 charset-normalizer-3.0.1 click-8.1.3 click-option-group-0.5.5 colorama-0.4.6 defusedxml-0.7.1 face-22.0.0 glom-22.1.0 idna-3.4 importlib-resources-5.10.2 jsonschema-4.17.3 packaging-21.3 parsy-2.0 peewee-3.15.4 pkgutil-resolve-name-1.3.10 pyparsing-3.0.9 pyrsistent-0.19.3 python-lsp-jsonrpc-1.0.0 requests-2.28.2 ruamel.yaml-0.17.21 ruamel.yaml.clib-0.2.7 semgrep-1.8.0 tomli-2.0.1 tqdm-4.64.1 typing-extensions-4.4.0 ujson-5.7.0 urllib3-1.26.14 wcmatch-8.4.1 zipp-3.12.0
WARNING: You are using pip version 22.0.4; however, version 23.0 is available.
You should consider upgrading via the '/private/tmp/semgrep7032/venv/bin/python3.8 -m pip install --upgrade pip' command.
$ venv/bin/pip show semgrep
Name: semgrep
Version: 1.8.0
Summary: Lightweight static analysis for many languages. Find bug variants with patterns that look like source code.
Home-page: https://github.com/returntocorp/semgrep
Author: Return To Corporation
Author-email: support@r2c.dev
License: UNKNOWN
Location: /private/tmp/semgrep7032/venv/lib/python3.8/site-packages
Requires: attrs, boltons, click, click-option-group, colorama, defusedxml, glom, jsonschema, packaging, parsy, peewee, python-lsp-jsonrpc, requests, ruamel.yaml, tomli, tqdm, typing-extensions, urllib3, wcmatch
Required-by:
```

I only get the latest version downloaded and it installs successfully.

I'm using MacOS on an Intel processor -- I'll try to find a colleague to try Linux or ARM.

**Comment by bmahe** [comment]
Hi!
We just released a new version of semgrep (1.9.0) which should resolve that problem.
Please try again and let us know!

**Comment by anotherbridge** [comment]
@bmahe Thanks very much for the quick update. I do not encounter any problems with the new release. So to me the issue is fixed.
